code/datums/spells/conjure.dm

```diff
@@ -13,16 +13,57 @@
 	var/list/newVars = list() //vars of the summoned objects will be replaced with those where they meet
 	//should have format of list("emagged" = 1,"name" = "Wizard's Justicebot"), for example
 
-	var/cast_sound = 'sound/items/welder.ogg'
-
 /obj/effect/proc_holder/spell/aoe_turf/conjure/cast(list/targets)
+	for(var/turf/T in targets)
+		if(T.density && !summon_ignore_density)
+			targets -= T
+
+	for(var/i=0,i<summon_amt,i++)
+		if(!targets.len)
+			break
+		var/summoned_object_type = pick(summon_type)
+		var/spawn_place = pick(targets)
+		if(summon_ignore_prev_spawn_points)
+			targets -= spawn_place
+		if(ispath(summoned_object_type,/turf))
+			if(istype(get_turf(usr),/turf/simulated/shuttle))
+				usr << "<span class='danger'>You can't build things on shuttles!</span>"

```

Comment:
That is referring to the Conjuration for Artifactors for Cult, which can build Walls, R-Walls, and flooring on majority of turf. Without that check they can also build on the shuttle turfs.